Improvement/upstream updates

jsf9k opened the pull request:
## 🗣 Description

This pull request adds types for all variables.  I also pull in upstream updates while I'm at it.

## 💭 Motivation and Context

I am going through all the Terraform module repositories to remove non-`alias` variables from proxy providers.  This repository did not require that change, but I noticed that the variable types were not always specified.

## 🧪 Testing

All pre-commit hooks pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

felddy committed:
Autoupdate pre-commit hooks.  Add mypy.

felddy committed:
Merge pull request #35 from cisagov/improvement/static_type_checking

Autoupdate pre-commit hooks.  Add mypy.

jsf9k committed:
Add .mypy_cache to .gitignore

jsf9k committed:
Merge pull request #36 from cisagov/improvement/add-mypy-cache-to-gitignore

Add .mypy_cache to .gitignore

mcdonnnj committed:
Incorporate the Python version into keys for pip and pre-commit caches.
This should resolve the issue seen when the Python version changes before there
is an update to .pre-commit-config.yml which results in pre-commit pointing to
a non-existent Python installation.

mcdonnnj committed:
Merge pull request #37 from cisagov/add_python_version_to_cache_keys

Incorporate the Python version into keys for pip and pre-commit caches.

mcdonnnj committed:
Change the cache paths from hardcoded values in their appropriate blocks to environment variables declared before the job block.

mcdonnnj committed:
Merge pull request #38 from cisagov/change_cache_paths_to_env_variables

Change Cache Paths to Environment Variables

mcdonnnj committed:
Add a rule for markdownlint to allow multiple headers with the same name as long
as they are not nested in the same heading group.

mcdonnnj committed:
Merge pull request #39 from cisagov/add_mdl_rule_for_MD024

Add markdownlint Configuration Option for MD024

mcdonnnj committed:
Fixed broken hook id for pre-commit-terraform:
  terraform_validate_no_variables was changed to terraform_validate in the
  following commit:
  https://github.com/antonbabenko/pre-commit-terraform/commit/35e0356188b64a4c5af9a4e7200d936e514cba71.
Ran pre-commit autoupdate.

mcdonnnj committed:
Merge pull request #40 from cisagov/improvements/update_pre-commit

Update pre-commit Hooks and Fix Broken Hook ID

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

mcdonnnj committed:
Add default provider information for the module.
Add Terraform initialization to the GitHub Actions workflow.

mcdonnnj committed:
Switch where the Terraform initialization takes place on the GitHub Actions
instance.

mcdonnnj committed:
We need to initialize the module and the example.

mcdonnnj committed:
Use my fork of the pre-commit-terraform hooks to allow the terraform_validate
hook to work with our directory tree.

mcdonnnj committed:
Adjust GitHub Actions workflow to eliminate superfluous 'cd ../..'.

mcdonnnj committed:
More default provider information for the module out of main.tf and into the
providers.tf per excellent feedback.

mcdonnnj committed:
Adjust 'terraform init' commands in the GitHub Actions workflow to eliminate
changing directories.

mcdonnnj committed:
+Add caching of .terraform directory to preserve provisioners between runs.
+Add directive to find and initialize Terraform directories.
-Removed directive that searched for Terraform directories.

mcdonnnj committed:
Switch back from personal fork of pre-commit-terraform now that my PR has been
merged.

mcdonnnj committed:
Ran `pre-commit autoupdate` to get the latest version of the
`pre-commit-terraform` hook since the PR I submitted,
https://github.com/antonbabenko/pre-commit-terraform/pull/100,
was approved. This will fix issues with `skeleton-tf-module` related to multiple
directories with Terraform code.

mcdonnnj committed:
Explicitly install pre-commit hooks as its own step so it's clearer when the
failure is with setup rather than hooks running.

mcdonnnj committed:
Rename action to accurately reflect that we are only linting, not building.
Adjust description for pre-commit hook setup to better convey what is being
done.

mcdonnnj committed:
Sort .gitignore entries.

mcdonnnj committed:
Merge pull request #41 from cisagov/bugfix/update_pre-commit_for_pre-commit-terraform_hook

Update pre-commit Hooks to Include Bugfix

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

mcdonnnj committed:
Merge pull request #16 from cisagov/improvements/upstream_updates

Pull in Upstream Changes

mcdonnnj committed:
Switch the "aws_region" in providers.tf to use the variable defined in
variables.tf. Add default value and type to "aws_region" and
"aws_availability_zone" variables. Add type to "subnet_id" variable.

mcdonnnj committed:
Add "-upgrade=true" to Terraform initialization in the GitHub Actions workflow.

mcdonnnj committed:
Add note to README that mentions requiring a `terraform init` in any directory
containing Terraform code to pass `pre-commit`.

mcdonnnj committed:
Fix wording of Notes section in README per feedback.

mcdonnnj committed:
Merge pull request #17 from cisagov/improvements/cleanup_README_and_default_configuration

Fix Default Configuration and Add Note to README

hillaryj committed:
Update CODEOWNERS

hillaryj committed:
Merge pull request #42 from cisagov/hillaryj-owner

Update CODEOWNERS

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-changes

jsf9k committed:
Merge pull request #18 from cisagov/improvement/upstream-changes

Pull in upstream changes from cisagov/skeleton-generic

felddy committed:
Allow events from apb to rebuild this repository weekly.

felddy committed:
Merge pull request #43 from cisagov/improvement/apb-events

Allow events from apb to rebuild this repository weekly.

dav3r committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_upstream_changes

dav3r committed:
Use latest version of GitHub action for Go and latest stable version of Go.

dav3r committed:
Explicitly turn on GO111MODULE environment variable.

dav3r committed:
Merge pull request #19 from cisagov/improvement/merge_upstream_changes

Fix terraform-docs install, update Go, merge upstream changes

mcdonnnj committed:
Updated pre-commit configuration with 'pre-commit autoupdate'.

felddy committed:
Add Lineage configuration.

felddy committed:
Merge pull request #20 from cisagov/improvement/lineage

Add Lineage configuration.

mcdonnnj committed:
Update isort pre-commit hook source repository.

The mirrors-isort respository has been deprecated per this Pull Request:
https://github.com/pre-commit/mirrors-isort/pull/13
Since isort includes pre-commit configuration in current versions we will
switch to using it directly.

mcdonnnj committed:
Ran pre-commit autoupdate.

mcdonnnj committed:
Remove legacy comment from the isort hook declaration.

This comment references the mirrors-isort repository that we are no longer
using.

mcdonnnj committed:
Disable the terraform_validate hook.

We have seen a number of issues related to this hook ever since it was
re-enabled. It will need to remain disabled until at least the 0.13 Terraform
release, and can only be re-enabled if all issues we have seen have been
resolved in how `terraform validate` operates.

mcdonnnj committed:
Merge branch 'develop' into improvements/pre-commit_autoupdate_2020-05

mcdonnnj committed:
Add final statement to descriptionf or why terraform_validate is disabled.

Review noticed that there lacked a determination for what we were doing about
the problem with the terraform_validate hook. I described the problems but
failed to mention what our path forward would be. This commit rectifies that
oversight.

mcdonnnj committed:
Merge branch 'improvements/pre-commit_autoupdate_2020-05' of github.com:cisagov/skeleton-generic into improvements/pre-commit_autoupdate_2020-05

mcdonnnj committed:
Merge pull request #44 from cisagov/improvements/pre-commit_autoupdate_2020-05

Update pre-commit Hook Versions

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

felddy committed:
Merge pull request #21 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

jsf9k committed:
Merge remote-tracking branch 'skeleton-tf-module/develop' into improvement/upstream-updates

jsf9k committed:
Update README via terraform-docs

dav3r commented on .github/lineage.yml:
```diff
@@ -0,0 +1,6 @@
+
+lineage:
+  skeleton:
+    remote-url: https://github.com/cisagov/skeleton-generic.git
```
This should be `skeleton-tf-module`, not `skeleton-generic`.

jsf9k replied:
See commit 827372d.

dav3r reviewed (approved):
Good to go once you fix the incorrect skeleton in `.github/lineage.yml`. 🚀

jsf9k committed:
Specify correct parent repo

jsf9k committed:
Add more directories where terraform init must be run in "Notes" section

jsf9k committed:
Fix typo

mcdonnnj committed:
Remove .terraform caching for GitHub Actions.

I originally added this caching to reduce the need to download providers for
Terraform initialization. However, when used with the `-upgrade=true` switch,
Terraform will always download the latest version of a provider without trying
to compare versions. Since it will always download the providers, there's no
point in caching them.

mcdonnnj committed:
Add debugging output to the loop the initializes Terraform directories.

I was looking at the output of a failing GitHub Actions job and realized that
the loop doesn't give an indication of what directory it's trying to initialize.
I think it's a good idea for debugging to know what has succeeded and where it
is failing for troubleshooting purposes, so I'm adding an `echo` to indicate
where it's trying to work.

mcdonnnj committed:
Removed erroneous characters from Terraform initialization loop.

When looking at GitHub Actions output from the previous commit, I saw that the
'[[' and ']]' are being treated as list items for Terraform initialization. This
is neither intended nor expected and so I am removing those characters from the
loop's `for` declaration.

mcdonnnj committed:
Add missing semicolon to echo in Terraform initialization loop.

Realized from the output that I'd forgotten this step as well.

mcdonnnj committed:
Merge pull request #22 from cisagov/remove_terraform_directory_caching

Remove GitHub Actions Caching for the '.terraform' Directory

jsf9k committed:
Merge remote-tracking branch 'skeleton-tf-module/develop' into improvement/upstream-updates

mcdonnnj reviewed (approved):
I'm not sure what happened to my previous review. You took care of the one thing I mentioned so LGTM ✔
